zaza/utilities/openstack.py

```diff
@@ -1220,7 +1238,15 @@ def get_os_release(release_pair=None):
     if release_pair is None:
         release_pair = get_current_os_release_pair()
     try:
-        index = OPENSTACK_RELEASES_PAIRS.index(release_pair)
+        try:
+            release_codename = release_pair.split('_')[-1]
+        except AttributeError:
+            release_codename = release_pair
+        ceph_releases = list(CEPH_CODENAMES.values())
+        if release_codename in ceph_releases:
```

Comment:
I'm slightly uncomfortable with this release comparison being the same for ceph + openstack, but there's not really a sane way to handle this otherwise. In this version, it's possible to do something like this:

```python3
bionic_mimic = openstack_utils.get_os_release('bionic_mimic')
xenial_newton = openstack_utils.get_os_release('xenial_newton')

assert(bionic_mimic < xenial_newton)
```

While that would be a very strange comparison (apples to oranges), it is technically possible given this implementation. My concern with splitting out the Ceph version comparison is that we really want to be able to compare OpenStack and Ceph version in the same way, if at all possible.

A potential way to mitigate this issue would be to create a dedicated comparison function so that we could raise an `Exception` in the event of mismatched entities (OS / Ceph) but that would require dramatically changing how we compare these versions.